### ISSUE TITLE
feat: implement PyTorch data pipeline

### ISSUE THREAD

**Issue by UstinAna** [UstinAna/breast-ultrasound-cnn#4]
## Summary

Implement the PyTorch data pipeline for loading, preprocessing, splitting, and batching ultrasound images.

## Changes

- Implemented custom PyTorch Dataset
- Added image preprocessing
- Added image resizing
- Added tensor conversion and normalization
- Implemented train/validation/test splitting
- Added reproducible random seed
- Implemented DataLoaders
- Added class distribution checks
- Added dataset tests
- Added visualization of processed samples

## Data Pipeline

Raw Images → Preprocessing → Dataset Split → DataLoader → Model

## Validation

- [ ] Dataset returns valid images and labels
- [ ] Image tensors have the expected shape
- [ ] Labels map correctly to the three classes
- [ ] Train/validation/test split is reproducible
- [ ] Test data is isolated from training
- [ ] DataLoaders produce valid batches
- [ ] Dataset tests pass
